src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -592,13 +604,23 @@ object AppDefinition extends GeneralPurposeCombinators {
     isTrue("Health check port indices must address an element of the ports array or container port mappings.") {
       case hc: HealthCheckWithPort =>
         hc.portIndex match {
-          case Some(PortReference.ByIndex(idx)) => hostPortsIndices.contains(idx)
-          case Some(PortReference.ByName(name)) => false // TODO(jdef) support port name as an index
+          case Some(mesosphere.marathon.core.health.PortReference.ByIndex(idx)) => hostPortsIndices.contains(idx)
+          case Some(mesosphere.marathon.core.health.PortReference.ByName(name)) => false // TODO(jdef) support port name as an index
           case None => hc.port.nonEmpty || (hostPortsIndices.length == 1 && hostPortsIndices.head == 0)
         }
       case _ => true
     }
 
+  private def checkPortIndexIsValid(hostPortsIndices: Range): Validator[Check] =
```

Comment:
will this work without using this method in `validBasicAppDefinition` 🤔 I am not sure and I think we definitely need test to be sure.

Reply:
it is in `validBasicAppDef` perhaps this was a comment from an early review.